osc-server/src/main/java/org/osc/core/broker/service/servicefunctionchain/AddServiceFunctionChainService.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service.servicefunctionchain;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.EntityManager;
+
+import org.osc.core.broker.model.entities.appliance.VirtualSystem;
+import org.osc.core.broker.model.entities.virtualization.ServiceFunctionChain;
+import org.osc.core.broker.model.entities.virtualization.VirtualizationConnector;
+import org.osc.core.broker.service.ServiceDispatcher;
+import org.osc.core.broker.service.api.AddServiceFunctionChainServiceApi;
+import org.osc.core.broker.service.dto.ServiceFunctionChainDto;
+import org.osc.core.broker.service.dto.VirtualSystemDto;
+import org.osc.core.broker.service.exceptions.VmidcBrokerValidationException;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.persistence.ServiceFunctionChainEntityMgr;
+import org.osc.core.broker.service.persistence.VirtualizationConnectorEntityMgr;
+import org.osc.core.broker.service.request.AddOrUpdateServiceFunctionChainRequest;
+import org.osc.core.broker.service.response.BaseJobResponse;
+import org.osc.core.broker.service.validator.DtoValidator;
+import org.osc.core.broker.service.validator.ServiceFunctionChainDtoValidator;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component
+public class AddServiceFunctionChainService
+		extends ServiceDispatcher<AddOrUpdateServiceFunctionChainRequest, BaseJobResponse>
+		implements AddServiceFunctionChainServiceApi {
+
+	DtoValidator<ServiceFunctionChainDto, ServiceFunctionChain> validator;
+	@Reference
+	private ServiceFunctionChainDtoValidator validatorFactory;
+
+	// private static final Logger LOG =
+	// Logger.getLogger(AddServiceFunctionChainService.class);
+
+	@Override
+	public BaseJobResponse exec(AddOrUpdateServiceFunctionChainRequest request, EntityManager em) throws Exception {
```

Comment:
need to add multiple validations

1. If no SDN controller, block creating SFC
2. If SDN controller does not support neutron SFC, throw an error

Reply:
Added TODO..Will modify as part of Install inspection hook story